src/main/java/seedu/timetable/Timetable.java

```diff
@@ -16,65 +16,65 @@
 
     private ArrayList<Module> modules;
 
-    private Class[] monday = new Class[24];
-    private Class[] tuesday = new Class[24];
-    private Class[] wednesday = new Class[24];
-    private Class[] thursday = new Class[24];
-    private Class[] friday = new Class[24];
-    private Class[] saturday = new Class[24];
-    private Class[] sunday = new Class[24];
+    private TimetableLesson[] monday = new TimetableLesson[24];
+    private TimetableLesson[] tuesday = new TimetableLesson[24];
+    private TimetableLesson[] wednesday = new TimetableLesson[24];
+    private TimetableLesson[] thursday = new TimetableLesson[24];
+    private TimetableLesson[] friday = new TimetableLesson[24];
+    private TimetableLesson[] saturday = new TimetableLesson[24];
+    private TimetableLesson[] sunday = new TimetableLesson[24];
 
```

Comment:
preferably use Arraylist for convenience methods, no need to track index

Reply:
I think its preferable to use fixed-size array where each index refers to the hour. i.e 23rd index is 2300, index 9 is 0900. This would make it easier to add timetable items to the array based on its starttime.

Reply:
This would also help make it easier during the add Module process when checking for collisions. Instead of using linear search to check for collisions can simply use the index to check if another module already occupies that time slot in O(1) time

Reply:
ohhh, i thought the 24 was for like max 24 items lol.
so now each timetable item does not store time anymore, since its tracked by array?
need to check if how gson handles null values in json array

Reply:
Hm the way I set it up was that the `TimetableLesson` is created first then added based on its `startTime` to `endTime` so I still have them as private properties that are accessed when adding to the timetable.
Ah I didn't consider the gson parsing, perhaps can initialize with some other starting value.

Reply:
looks like gson can handle null values (so something like [1,2, null, null, 3])
lets just go with ur implementation